### PR TITLE
sixth homework

### PR DESCRIPTION
**Pull request by natisun77**

### CODE REVIEW

File: src/homework/homeworks/Animal.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+public class Animal implements Serializable {
+    private final String name;
+
+    public Animal(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```
Comment (Siril-O) [inline src/homework/homeworks/Animal.java:20]: override hash code

File: src/homework/homeworks/Animal.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                Object obj = inputReader.readObject();
+                if (obj instanceof Animal) {
+                    animals[i] = (Animal) obj;
+                } else {
+                    throw new IllegalArgumentException();
+                }
+            }
+            return animals;
+        } catch (IOException | ClassNotFoundException e) {
+            throw new IllegalArgumentException();
```
Comment (Siril-O) [inline src/homework/homeworks/Animal.java]: wrap e to IllegalArgumentException or print stacktrace

File: src/homework/homeworks/Animal.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            for (int i = 0; i < sizeOfArray; i++) {
+                Object obj = inputReader.readObject();
+                if (obj instanceof Animal) {
+                    animals[i] = (Animal) obj;
+                } else {
+                    throw new IllegalArgumentException();
+                }
+            }
+            return animals;
+        } catch (IOException | ClassNotFoundException e) {
```
Comment (Siril-O) [inline src/homework/homeworks/Animal.java:54]: e is not valid name

File: src/homework/homeworks/Animal.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        Animal Dog = new Animal("Dog");
+        Animal Cat = new Animal("Cat");
+        Animal Horse = new Animal("Horse");
+        Animal Zebra = new Animal("Zebra");
+        Animal[] animals = new Animal[]{Dog, Cat, Horse, Zebra};
+
+        byte[] arrayOfBytes = serializeAnimals(animals);
+        String result = Arrays.toString(deserializeAnimalArray(arrayOfBytes));
+        System.out.println(result);
+        //     Animal[] array = deserializeAnimalArray(new byte[5]);
```
Comment (Siril-O) [inline src/homework/homeworks/Animal.java]: remove commented out line

File: src/homework/homeworks/SetExample.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
+package homework.homeworks;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class SetExample {
+
+    public static <T> Set<T> symmetricDifference(Set<? extends T> set1, Set<? extends T> set2) {
+        Set<T> combinedSetFirst = new HashSet<>();
```
Comment (Siril-O) [inline src/homework/homeworks/SetExample.java]: this can be done easier

**Review by Siril-O — APPROVED** [review]

File: src/homework/homeworks/Animal.java (hunk 3 above)
Comment (natisun77) [reply src/homework/homeworks/Animal.java:54]: As I know if often used in java and saw a lot of examples. Is not correct to leave it like this? What is should be?

**Pull request merged**